docs/rfcs/XXXX_storage_backends.md

```diff
@@ -0,0 +1,248 @@
+
+### New Layout
+```
+studies/
+  SD_00000000/
+    source/
+    uploads/
+    harmonized/
+```
+
+## External Storage Backends
+
+![enter image description here](https://i.ibb.co/PTPMLnf/storage-layout.png)
+Under circumstances that an organization requires all of their uploads or data files to be under their control, an organization administrator may specify credentials and details for a bucket of their choosing in a different account.
+
+### Storage Backends
+
+A Storage Backend is a simple entity that belongs to an organization and specifies IAM user credentials, a bucket that the IAM user has access to, and a prefix within that bucket. This concept will allow Dewrangle to access buckets across different accounts for the purpose of storing or evaluating files.
+
+### Configuration
+To configure a storage backend, an organization administrator will be required to provide details about the AWS account where their target bucket resides and a user who may access it. An organization administrator will be able to configure this by navigating to the *Organization Settings* screen and choosing *Storages*. This will allow the administrator to add a new storage by first asking the user to create a new IAM user in their AWS account and assigning a specified policy. After the user has done this, they will be required to provide the AWS access and secret key pair so that Dewrangle may access the bucket. The final step will be to allow the user to specify the bucket and what prefix under it they would like this storage backend to use as the root. After confirming their choices, Dewrangle will attempt to access the bucket with the provided credentials and save the new storage backend if successful.
+
+### Use of Storage Backends
+
+A Storage Backend may be used in one of two ways: Either as a Document Storage Backend or a Data Storage Backend. Each method will require the same configuration of a Storage Backend and, in fact, a single Storage Backend may be used for both Documents and Data.
+
+#### Document Storage Backends
+
+Storage Backends used for documents will be read/write. Dewrangle will use these backends to save documents using a `uploads/` prefix under the root of the Storage Backend.
+
+#### Data Storage Backends
+
+Storage Backends used for data will only utilize read and list operations. These backends will be used for file discovery and accounting only.
+
+### Setting Organization Storage Backend Defaults
+
+If an organization administrator wishes all new studies to use a given backend for document uploads or data files, they may specify it in the organization settings. The administrator will have two defaults to configure: *Default Document Storage Backend* and *Default Data Storage Backend*. An administrator may use the same Storage Backend for both, but it will be up to them to prevent collisions.
+
+### Changing Storage Backends Per-Study
```

Comment:
I believe I address the first question somewhere in here. The answer is that it is the responsibility of the user to make those decisions and contact us if they need help migrating old files.

For the second question, yes, the suggested method here would be create a new storage backend with a more specific prefix. This should not be encouraged though because a single storage backend will automatically prefix studies in its own way.

The uniqueness should probably be constrained to org+bucket+prefix. I don't see a reason to have the same org+bucket+prefix but using a different user, but I will have to think about it some more.